:video_game: Improved 'LiveRepair' (formerly 'interactive repair')

only-a-ptr opened the pull request:
I renamed the feature because this is shorter and fits better in UIs and variable names. Hopefully it'll also fit better in player's vocabulary. Fixes #3062

**Changes:**
* new UI box "Live repair active" with optional display of hotkeys. It displays right under the top menubar (like existing "physics paused/terrain editor" boxes). See screenshot below.
* new input event EV_COMMON_LIVE_REPAIR_MODE (default Alt+Backspace) - throws you right into the LiveRepair mode, no need to wait.
* new cvar 'sim_live_repair_interval' (float) - default 2 sec - how long you must hold Backspace and be still before the LiveRepair mode kicks in. Used to be 1sec.
* new cvar 'ui_show_live_repair_controls' (bool) - controls the UI checkbox in "Live repair is active" statebox. Also configurable in GameSettings UI.

![obrazek](https://github.com/RigsOfRods/rigs-of-rods/assets/491088/25816b42-d29b-4037-9f7e-e40853f02ad3)

**New (juicy) helper functions:**
* `void ImDrawEventHighlighted(events input_event)` - draws the configured key combo for given input event, with a darker frame around it and color highlight when the keys are pressed.
* `void ImDrawModifierKeyHighlighted(OIS::KeyCode key)` - draws the localized modifier key name (shift/alt/ctrl), with color highlight when the keys are pressed.
* `InputEngine::getModifierKeyName(OIS::KeyCode key)` - gives you localized name for the modifier key (shift/alt/ctrl)

CuriousMike56 commented:
Not sure if intentional, but the hotkey display doesn't appear if you hold backspace during live repair. Works good otherwise 👍

only-a-ptr commented:
**Updated new vocabulary:**
* **QuickRepair** = press and hold `EV_COMMON_REPAIR_TRUCK` (default Backspace), release to stop. The notify box title is orange because it's a press-and-hold state.
* **LiveRepair** = hit `EV_COMMON_LIVE_REPAIR_MODE` (default Alt+Backspace) or keep the quickrepair pressed for 'sim_live_repair_interval' seconds (default 2). The notify box title is green because it's a persistent state, like the terrain editor or physics pause.

Both have identical controls and show the same notify box, but with slightly different title, and progressbar for the 'sim_live_repair_interval'

**Other changes:**
* Fixed UpDown and ForwardReverse being swapped in the notifiy box
* Fixed 'sim_live_repair_interval' not recognizing 0 as "do nothing" - potential division by zero.
* Visual: Fixed missing frame around modifier keys
* Improvement: added indent to keys in the notify box.
* Added reset mode (soft/hard) to notify box
